apps/aecore/test/aec_chain_state_tests.erl

```diff
@@ -662,6 +663,25 @@ fork_common_headers(EasyChain, TopHashEasy, HardChain, TopHashHard) ->
     ?assertEqual(TopHashHard, top_header_hash(State4)),
     ok.
 
+fork_out_of_order() ->
+    CommonChain = gen_block_chain_by_target([?GENESIS_TARGET, 1, 1], 111),
+    EasyChain = extend_chain(CommonChain, [2], 111),
+    HardChain = extend_chain(CommonChain, [1], 222),
+
+    %% Add the chain with the fork node as the last entry.
+    InitState = new_state(),
+
+    State1 = write_blocks_to_chain(butlast(CommonChain), InitState),
+    {ok, State2} = insert_block(lists:last(EasyChain), State1),
+    {ok, State3} = insert_block(lists:last(HardChain), State2),
+
+    %% The last block to enter is the last common node.
+    ?assertMatch({ok, _}, insert_block(lists:last(CommonChain), State3)),
```

Comment:
I was expecting checking new top here - though I am still getting understanding of PR.

Reply:
Agreed. The symptom before was to crash, though.

I'll add the test.

Reply:
If I keep test but remove code change I see the badmatch - thanks.

Reply:
I saw you added the checks in force-push - thanks. You have been faster than me!

Reply:
I also tested locally with additional checks on top (header and block) at the several intermediate stages, all ok.